CMakeLists.txt

```diff
@@ -10,6 +10,7 @@ endif()
 option(LIBSAMPLERATE_TESTS "Enable to generate test targets" ${IS_ROOT_PROJECT})
 option(LIBSAMPLERATE_EXAMPLES "Enable to generate examples" ${IS_ROOT_PROJECT})
 option(LIBSAMPLERATE_INSTALL "Enable to add install directives" ${IS_ROOT_PROJECT})
+option(LIBSAMPLERATE_ENABLE_WARNINGS "Enable compiler warnings and fail compile if any triggers" OFF)
```

Comment:
I think disabling warnings by default is a bad idea. I agree they should never be promoted to errors by default, but all warnings should be visible.

Even if warnings were enabled by default I would question the wisdom of being able to disable them.

Reply:
Then I propose 2 options: LIBSAMPLERATE_ENABLE_WARNINGS, LIBSAMPLERATE_ENABLE_WERROR. Alternatively just the latter.

Reason for having an option with default OFF is that I set those as INTERFACE option which enables them on all targets linking to this. If I remove LIBSAMPLERATE_ENABLE_WARNINGS and just add LIBSAMPLERATE_ENABLE_WERROR then I'd add warnings per target (PRIVATE for samplerate, tests and examples can be PUBLIC, much easier with #69 as there is a common library used by all tests/examples respectively, so only repeat them 3 times)

BTW: There are some warnings (especially on Windows)  about float->double conversions which I disabled here. Better would be to declare constants with `f` suffix and add explicit casts to `out[k] = ...` assignments which also makes the conversion visible and is good IMO, but you said you want to avoid casts.

Reply:
Having `LIBSAMPLERATE_ENABLE_WERROR` would mirror the autotool config option `--enable-werrror`.

> There are some warnings (especially on Windows) about float->double conversions

I think there is a MSVC `#pragma` to disable those.

Reply:
So drop `LIBSAMPLERATE_ENABLE_WARNINGS` and add `LIBSAMPLERATE_ENABLE_WERROR`?

> I think there is a MSVC #pragma to disable those.

So add `#pragma push/disable/pop` to the code instead of conversions? I'd advice against that but your call

Reply:
Yes, enable warnings by default, enable `-Werror` as an option.

Yes, I would prefer the `#prama` fix than adding casts.

Reply:
Done. As warnings are now always enabled I check that the compiler does not error out if the flag is passed (check_c_compiler_flag)

Also done. Used the MSVC definition to only add them for MSVC to avoid e.g. `-Wunknown-pragma` warnings. I'm not particularly fond of suppressing the warnings as IMO narrowing conversions shouldn't be done implicitly as loss of data happens.